service/resource/cloudformation/resource.go

```diff
@@ -0,0 +1,111 @@
+package cloudformation
+
+import (
+	"context"
+
+	"github.com/giantswarm/certificatetpr"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/framework"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	// Name is the identifier of the resource.
+	Name = "cloudformation"
+)
+
+// Config represents the configuration used to create a new config map resource.
+type Config struct {
+	// Dependencies.
+	CertWatcher certificatetpr.Searcher
+	//CloudConfig *cloudconfig.CloudConfig
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+}
+
+// DefaultConfig provides a default configuration to create a new config map
+// resource by best effort.
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+		CertWatcher: nil,
+		//CloudConfig: nil,
+		K8sClient: nil,
+		Logger:    nil,
+	}
+}
+
+// Resource implements the config map resource.
+type Resource struct {
+	// Dependencies.
+	certWatcher certificatetpr.Searcher
+	//cloudConfig *cloudconfig.CloudConfig
+	k8sClient kubernetes.Interface
+	logger    micrologger.Logger
+}
+
+// New creates a new configured config map resource.
+func New(config Config) (*Resource, error) {
+	// Dependencies.
+	if config.CertWatcher == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.CertWatcher must not be empty")
+	}
+	//if config.CloudConfig == nil {
+	//	return nil, microerror.Maskf(invalidConfigError, "config.CloudConfig must not be empty")
+	//}
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.K8sClient must not be empty")
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+
+	newService := &Resource{
+		// Dependencies.
+		certWatcher: config.CertWatcher,
+		//cloudConfig: config.CloudConfig,
+		k8sClient: config.K8sClient,
+		logger: config.Logger.With(
+			"resource", Name,
+		),
+	}
+
+	return newService, nil
+}
+
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+	return nil, nil
+}
```

Comment:
It would be cool to separate the steps into multiple files. We can do this in a separate PR though.

Reply:
Yep, this is coming here https://github.com/giantswarm/aws-operator/pull/484

Reply:
Yes, I like the separating into multiple files. Let's do in a later PR.